### ISSUE TITLE
no-unused-prop-types is confused by nested PropTypes

### ISSUE THREAD

**Issue by Wilfred** [yannickcr/eslint-plugin-react#1163]
```js
import React from "react";

var Hello = React.createClass({
  propTypes: {
    name: React.PropTypes.string,
    foo: React.PropTypes.string,
    propTypes: React.PropTypes.string
  },
  render: function() {
    return <div>Hello {this.props.name}</div>;
  }
});
```

Produces:

```bash
$ eslint --no-eslintrc --parser-options="{sourceType: \"module\", ecmaVersion: 6, ecmaFeatures: {jsx:true}}" --plugin react --rule react/no-unused-prop-types:error js/Hello.jsx
# No errors produced.
```

I agree this is a dumb bit of code, but it would be nice to robust. I had some broken copy-pasted code of the form:

```js
var Hello = React.createClass({
  propTypes: {
    name: React.PropTypes.string,
    propTypes: {
      foo: React.PropTypes.string,
    }
  },
  render: function() {
    return <div>Hello {this.props.name} {this.props.foo}</div>;
  }
});
```

**Comment by ljharb** [comment]
React.createClass is officially deprecated as of React v15.5; we should still fix the former case tho.

**Comment by ljharb** [comment]
Looks like the issue is you're missing `settings: { react: { createClass: 'createClass' } }` in your eslint config; with that, it passes fine.
